lib/source_file.h

```diff
@@ -261,6 +266,11 @@ class Comment final : IHasDbPrint {
         out << body;
         if (!singleLine) out << "*/";
     }
+
+    /// Retrieve the beginning of the source position associated with this comment.
+    [[nodiscard]] const SourcePosition &getStartPosition() const { return srcInfo.getStart(); }
+    /// Retrieve the end of the source position associated with this comment.
+    [[nodiscard]] const SourcePosition &getEndPosition() const { return srcInfo.getEnd(); }
```

Comment:
It would be more natural to me to expose the source info of the comment. That is for it to be `IHasSourceInfo`. You may still expose the start and end if you think it is necessary, but I suggest Comment behaves more like other things in this way.